src/Notifier.js

```diff
@@ -123,6 +123,26 @@ var Notifier = {
         return global.Notification.permission == 'granted';
     },
 
+    isPermissionDefault: function() {
+        if (!this.supportsDesktopNotifications()) return false;
+        return global.Notification.permission == 'default';
+    },
+
+    showToolbar: function() {
+        // Check localStorage for any such meta data
+        if (global.localStorage) {
+            if (global.localStorage.getItem('notifications_hidden') === 'true')
+                return false;
+        }
+
+        // Check if permission is granted by any chance.
+        if (this.havePermission()) return false;
+
+        // means the permission is blocked
+        if (!this.isPermissionDefault()) return false;
+        return true;
```

Comment:
@richvdh Tell me what do you think about showing a message if the user has blocked notif request **& not yet attempted to close the toolbar** - if yes, I'll change this method to return enums - 'show', 'hide', 'blocked' and the vector will take decision accordingly.